src/monitoring.tbasicu

```diff
@@ -0,0 +1,167 @@
+Uses "Biff"
+
+#include once "logger_ps.tbasicu"
+
+%DIFFICULTY_MONITOR_RATE = 0.5 's
+%DIFFICULTY_STORE_RATE = 2 ' s (averaging)
+
+Type monitorData
+  temps as Double
+  difficulty as Double
+  meanFrameRate as dword
+  meanDistance as Double
+  meanCapacity as double
+end type
+
+
+Type difficultyMonitor
+
+  contextPtr as dword
+  logger as LoggerPS
+  difficulty as double
+  oldTime as Double ' to calculate time diff between two calls
+  tempsMonitor as double
+  tempsStore as Double
+  rateCount as DWord
+  '[t] can't use undimensioned array conjointly with " Local context as GameContext at me.contextPtr "
+  storage(200000) as monitorData
+  mesuresCount as DWord
+  oneData as monitorData
+  gamesessionStartTime as double
+  
+  function Prepare(contextPtr as dword)
+
+    me.contextPtr = contextPtr
+    Local context as GameContext at me.contextPtr
+
+    #if %LOGGERPS
+      me.logger.initialize("difficultyMonitor")
+    #endif
+    
+    me.gamesessionStartTime = context.gameTime
+    
+    redim me.storage(10)
+    me.mesuresCount = 0
+  
+  end function
+  
+  Function Close()
+  
+    Local context as GameContext at me.contextPtr
+    Local currentTime as double = context.gameTime
+  
+    me.Store(currentTime)
+    me.exportToFile
+    Redim me.storage(1)
+    
+  end function
+
+  
+  Function Rate()
+  
+    LOCAL x, y, z, distance, difficulty, capacity, counter AS Double
+    Local context as GameContext at me.contextPtr
+    Local totalDistance, totalCapacity as Double
+    
+    tbgl_entityGetPos(uScreenGame.player.scene,uScreenGame.player.entity,x,y,z,%TRUE) 'Absolute Player position
+    
+    for i as integer = 1 to uScreenGame.bubbleDirector.bubbleTotalCount
+      capacity = uScreenGame.bubbleDirector.bubble(i).capacity ' Bubble capacity
+      if capacity = 0 then iterate for
+      incr counter
+      distance = tbgl_entityGetDistancePos(uScreenGame.player.scene,uScreenGame.bubbleDirector.bubble(i).entity,x,y,z) ' Bubble---Player Distance
+      totalDistance += distance
+      totalCapacity += capacity
+      me.oneData.difficulty += POW3(1/capacity) * POW2(distance) ' One far big bubble is easier to catch than many close little ones 
+    next
+    
+    me.oneData.meanFrameRate += context.framerate
+    me.oneData.meandistance += totalDistance / counter
+    me.oneData.meanCapacity += totalCapacity / counter
+    counter = 0 
+  end function
+  
+  Function Monitor()
+'[breakpoint] <Any note here. Breakpoint will be set to line following this statement>
+'[t] on second pass me.contextPtr is lost (reset to 0 , why ? I think it's because storage() was redim. )
+    Local context as GameContext at me.contextPtr
```

Comment:
Yes, it is memory corruption indeed. It is a bug in thinBASIC.

I am trying to prepare minimal example for Eros, however, it seems to be quite tricky.